code/src/main/java/com/onlinecontacttracing/servlets/authentication/CheckForApiAuthorizationServlet.java

```diff
@@ -61,19 +61,27 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       String code = request.getParameter("code");
       String idTokenString = request.getParameter("state");
 
-      // Get userId
-      String userId = getUserId(idTokenString, flow, response);
+      // Get payload containing user info
+      Payload payload = getUserId(idTokenString, flow, response);
+
+      // Get userId form payload and retrieve credential
+      String userId = payload.getSubject();
+      String email = payload.getEmail();
+      
       TokenResponse tokenResponse = flow.newTokenRequest(code).setRedirectUri(url+getServletURIName()).execute();
       Credential credential = flow.createAndStoreCredential(tokenResponse, userId);
-      updateUser(userId);
-      useCredential(credential);
+      updateUser(userId, email);
+      useCredential(credential, response);
+
     } catch (FileNotFoundException e) {
       log.warning("credentials.json not found");
       response.sendRedirect("/?page=login&error=FileError");
     } catch (GeneralSecurityException e) {
       log.warning("http transport failed, security error");
-      response.sendRedirect("/?page=login&error=TransportError");
-    } catch(Exception e) { //don't expect any other error
+      response.sendRedirect("/?page=login&error=GeneralError");
+    } catch (InterruptedException e) {
```

Comment:
If you're not handling it at all different from the general exception it doesn't make sense to add a new catch.

Did you intend to log something here? Or redirect to a different page?

Reply:
No, I just wanted to show it was handled but I can remove it.

Reply:
Cool

Reply:
Okay, I removed it